controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,205 @@
+	odfInfoConfigOverrideData := map[string]string{
+		odfInfoKey: odfInfo,
+	}
+	odfInfoConfigOverrideCM := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       RDROdfInfoMapKind,
+			APIVersion: sc.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RDROdfInfoConfigMapName,
+			Namespace: sc.Namespace,
+		},
+		Data: odfInfoConfigOverrideData,
+	}
+
+	_, err := ctrl.CreateOrUpdate(context.Background(), r.Client, odfInfoConfigOverrideCM, func() error {
+		if !reflect.DeepEqual(odfInfoConfigOverrideCM.Data, odfInfoConfigOverrideData) {
+			r.Log.Info("updating ODF info config override configmap", RDROdfInfoMapKind, klog.KRef(sc.Namespace, RDROdfInfoConfigMapName))
+			odfInfoConfigOverrideCM.Data = odfInfoConfigOverrideData
+		}
+		return ctrl.SetControllerReference(sc, odfInfoConfigOverrideCM, r.Scheme)
+	})
+	if err != nil {
+		r.Log.Error(err, "failed to create or update ODF info config override", RDROdfInfoMapKind, klog.KRef(sc.Namespace, RDROdfInfoConfigMapName))
+		return reconcile.Result{}, fmt.Errorf("failed to create or update ODF info config override: %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.StorageCluster) (reconcile.Result, error) {
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoConfig(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (string, error) {
+	var odfVersion, cephFSId string
+	var err error
+	if odfVersion, err = getOdfVersion(r, sc); err != nil {
+		return "", err
+	}
+	if cephFSId, err = getCephFsid(r, sc); err != nil {
+		return "", err
+	}
+	var odfDeploymentType string
+	if sc.Spec.ExternalStorage.Enable {
+		odfDeploymentType = RDROdfDeploymentTypeExternal
+	} else {
+		odfDeploymentType = RDROdfDeploymentTypeInternal
+	}
+	var isDROptimized = false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !sc.Spec.ExternalStorage.Enable {
+		if isDROptimized, err = getIsDROptimized(r, sc); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(sc); err != nil {
+		return "", err
+	}
+	storageClusterCount := len(r.clusters.GetStorageClusters())
+
+	var data = OdfInfoData{
+		OdfVersion:          odfVersion,
+		OdfDeploymentType:   odfDeploymentType,
+		StorageClusterCount: storageClusterCount,
+		IsDROptimized:       isDROptimized,
+		StorageSystemName:   storageSystemName,
+		// Clients array is populated with the onboarding request's fields via server.go
+		Clients: []ConnectedClient{},
+		StorageCluster: InfoStorageCluster{
+			Metadata:                types.NamespacedName{Name: sc.Name, Namespace: sc.Namespace},
+			StorageProviderEndpoint: sc.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := json.Marshal(data)
```

Comment:
The variable is called `yamlData` while you are using `JSON` marshaling, which produces a JSON document, not a YAML document.  

Please use yaml merashling instead (which also means you will need to add the `yaml:"..."` tag to the structure fields.